sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/data/redis/AzureLettucePasswordlessAutoConfiguration.java

```diff
@@ -54,8 +66,12 @@ AzureRedisPasswordlessProperties redisPasswordlessProperties() {
     AzureRedisCredentials azureRedisCredentials(RedisProperties redisProperties,
                                                 AzureRedisPasswordlessProperties azureRedisPasswordlessProperties,
                                                 AzureGlobalProperties azureGlobalProperties) {
-        return new AzureRedisCredentials(redisProperties.getUsername(),
-            mergeAzureProperties(azureGlobalProperties, azureRedisPasswordlessProperties));
+        AzureRedisPasswordlessProperties redisPasswordlessProperties = mergeAzureProperties(azureGlobalProperties, azureRedisPasswordlessProperties);
+        String passwordlessPropertiesPrefix = "spring.data.redis.azure";
+        registerTokenCredentialBean(applicationContext, passwordlessPropertiesPrefix, redisPasswordlessProperties);
```

Comment:
Do we really need to register the bean here, if the token credential provider can actually resolve one according to options?